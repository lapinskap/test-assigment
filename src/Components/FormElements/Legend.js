import PropTypes from 'prop-types';
import React from 'react';
import __ from '../../utils/Translations';

const displayLegend = (legend) => {
  switch (legend) {
    case typeof (legend) === 'object':
      return legend.map((item) => (
        <p className="text-center font-weight-normal">{item}</p>
      ));
    case typeof (legend) === 'string':
      return <p>{legend}</p>;
    default:
      return '-';
  }
};

export default function Legend({
  id, title, legend, translateTitle,
}) {
  return (
    <div className="col-sm-12">
      <p id={id}>
        {translateTitle ? __(title) : title}
      </p>
      <div className="col-sm-12">
        {displayLegend(legend)}
      </div>
    </div>
  );
}

Legend.propTypes = {
  legend: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  translateTitle: PropTypes.bool,
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  id: PropTypes.string.isRequired,
};

Legend.defaultProps = {
  legend: [],
  translateTitle: true,
  title: 'Legenda:',
};
