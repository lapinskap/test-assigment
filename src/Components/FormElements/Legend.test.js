import React from 'react';
import renderer from 'react-test-renderer';
import Legend from './Legend';
import '../../tests/setupTests';

function createNodeMock() {
  return {
    getBoundingClientRect() {
      return {
        width: 100,
      };
    },
  };
}

describe('<Legend />', () => {
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');
  useStateSpy.mockImplementation((init) => [init, setState]);
  let title;
  let validateForm;

  beforeEach(() => {
    title = 'Test title';
    validateForm = () => {
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with all props', () => {
    const legend = [
      'id_test - test value 1',
      'id_test - test value 2',
      'id_test - test value 3',
    ];
    const tree = renderer
      .create(<Legend
        legend={legend}
        title={title}
        validateForm={validateForm}
      />, { createNodeMock })
      .toJSON();

    expect(tree)
      .toMatchSnapshot();
  });

  it('renders correctly with valid legend', () => {
    const legend = [
      'id_test - test value 1',
      'id_test - test value 2',
      'id_test - test value 3',
    ];
    const tree = renderer
      .create(<Legend
        legend={legend}
        title={title}
        stickyTitle
        validateForm={validateForm}
      />, { createNodeMock })
      .toJSON();

    expect(tree)
      .toMatchSnapshot();
  });
});
