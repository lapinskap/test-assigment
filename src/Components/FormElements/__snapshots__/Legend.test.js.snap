// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Legend /> renders correctly with all props 1`] = `
<div
  className="col-sm-12"
>
  <p>
    Test title
  </p>
  <div
    className="col-sm-12"
  >
    -
  </div>
</div>
`;

exports[`<Legend /> renders correctly with valid legend 1`] = `
<div
  className="col-sm-12"
>
  <p>
    Test title
  </p>
  <div
    className="col-sm-12"
  >
    -
  </div>
</div>
`;
